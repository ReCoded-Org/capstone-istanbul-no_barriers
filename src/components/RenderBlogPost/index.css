.profileBlogPostImage,
.imageHover {
  width: 20rem;
}

.profileBlogPostInfo {
  background-color: var(--blog-background-profile);
}

.blogPostInfoHover {
  color: var(--blog-background-profile);
  background-color: var(--main-text-color-profile);
}

.profileBlogPostHover {
  background-color: var(--main-text-color-profile);
  cursor: pointer;
}

.imageHover {
  opacity: 0.4;
}
