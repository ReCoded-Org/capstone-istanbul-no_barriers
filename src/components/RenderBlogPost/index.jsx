import React, { useState } from "react";
import { Image } from "react-bootstrap";
import "./index.css";

const RenderBlogPost = ({ postInfo }) => {
  const [hover, setHover] = useState(false);

  return (
    <div
      className={hover ? "profileBlogPostHover" : "pt-2 profileBlogPost"}
      onMouseEnter={() => setHover(!hover)}
      onMouseLeave={() => setHover(!hover)}
    >
      <Image
        src={postInfo.image}
        alt="postInfo"
        fluid
        className={hover ? "imageHover" : "profileBlogPostImage"}
      />
      <div
        className={
          hover
            ? "blogPostInfoHover px-2 pt-2"
            : "profileBlogPostInfo px-2 pt-2"
        }
      >
        <h5>{postInfo.title}</h5>
        <p className="pb-2">{postInfo.date}</p>
      </div>
    </div>
  );
};

export default RenderBlogPost;
