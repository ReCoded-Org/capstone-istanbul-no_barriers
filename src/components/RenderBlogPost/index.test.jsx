import React from "react";
import renderer from "react-test-renderer";
import RenderBlogPost from "./index";
import profilePic from "./profile.jpg";

const postInfo = { image: profilePic, title: "My title", date: "11/11/11" };
it("RenderBlogPost renders correctly", () => {
  const tree = renderer.create(<RenderBlogPost postInfo={postInfo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
