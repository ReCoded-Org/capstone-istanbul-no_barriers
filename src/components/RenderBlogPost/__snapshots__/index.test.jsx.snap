// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderBlogPost renders correctly 1`] = `
<div
  className="pt-2 profileBlogPost"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="postInfo"
    className="profileBlogPostImage img-fluid"
    src="profile.jpg"
  />
  <div
    className="profileBlogPostInfo px-2 pt-2"
  >
    <h5>
      My title
    </h5>
    <p
      className="pb-2"
    >
      11/11/11
    </p>
  </div>
</div>
`;
